website/static/js/notificationsTreebeard.js

```diff
@@ -139,6 +139,36 @@ function ProjectNotifications(data) {
                     }
                 });
             }
+            else if (item.parent().data.kind === 'folder' || item.parent().data.kind === 'heading' && item.data.kind === 'event' && item.data.event.notificationType === 'mailing_list_events') {
```

Comment:
Another complex condition. At a minimum we should use parens to make grouping explicit here, with a comment to explain what this condition is doing.